controller.go

```diff
@@ -469,3 +457,71 @@ func upgradePVFrom12(pv *PV) (deleted bool, err error) {
 	}
 	return false, nil
 }
+
+func provisionClaim(pvc *api.PersistentVolumeClaim) error {
+	// Provisioning can take some time. Start it in a separate goroutine.
+	// Keep a map of running provisioners and start new provisioning only if
+	// it's not already running. There will be some locks around, these are
+	// left out to implementation
+
+	if !runningProvisioners.isRunning(pvc) {
+		runningProvisioners.markAsRunning(pvc)
+		go doProvisionClaim(pvc)
+	}
+}
+
+func doProvisionClaim(pvc *api.PersistentVolumeClaim) {
+	// On exit, mark the provisioner for this claim as finished
+	defer runningProvisioners.markAsFinished(pvc)
```

Comment:
Is this the right thing to do?  This will mark the provisioner as finished no matter what happens in this method.